### PR TITLE
Feature/jaxrs2 reader oas v3.0.0

### PR DESCRIPTION
**Pull request by ralphdoe**
A new module was added (swagger-jaxrs2).
This is the beginning of the new OpenApi Spec 3.0 Object Construction. 
It includes the Operation Object construction and the Paths of the OpenApi Object.
There's a ReaderTest for Testing the functionalities that were added. 
It also has a lot of reusable functionalities for parsing the Annotations into Objects.

@fehguy

### CODE REVIEW

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+ */
+public class OperationParser {
+
+	public static Optional<List<Parameter>> getParametersList(io.swagger.oas.annotations.Parameter[] parameters) {
+		if (parameters == null) {
+			return Optional.empty();
+		}
+		List<Parameter> parametersObject = new ArrayList<>();
+		for (io.swagger.oas.annotations.Parameter parameter : parameters) {
+			Parameter parameterObject = new Parameter();
```
Comment (fehguy) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java]: I think this logic isn't right.  Parameters may be _explicit_ by being decorated with `@Schema` annotations, or with `@QueryParam`, etc.  Look at the `ParameterProcessor` logic from the previous version for more details on this.

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/Reader.java [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+			default:
+				// Do nothing here
+				break;
+		}
+	}
+
+	private void setOperationObjectFromApiOperationAnnotation(Operation operation, io.swagger.oas.annotations.Operation apiOperation) {
+		ReaderUtils.getStringListFromStringArray(apiOperation.tags()).ifPresent(tags -> operation.setTags(tags));
+		OperationParser.getTags(apiOperation.tags()).ifPresent(tag -> openApiTags.addAll(tag));
+		operation.setSummary(apiOperation.summary());
```
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/Reader.java]: StringUtils.isNotBlank validations added.

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/Reader.java [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+			Map<String, String> regexMap = new LinkedHashMap<>();
+			operationPath = PathUtils.parsePath(operationPath, regexMap);
+			if (operationPath != null) {
+				if (ReaderUtils.isIgnored(operationPath, config)) {
+					continue;
+				}
+
+				Operation operation = parseMethod(method);
+				PathItem pathItemObject = new PathItem();
+				pathItemObject.set$ref(operation.getOperationId());
```
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/Reader.java]: Changed.

**Comment by fehguy** [comment]
@ralphdoe this is looking good.  There are some issues still but I would like to merge this in and address updates as they come.

**Pull request merged**